src/EnergyPlus/UnitarySystem.hh

```diff
@@ -195,7 +195,9 @@ namespace UnitarySystems {
 
         UnitarySysInputSpec original_input_specs;
         int m_UnitarySysNum;
-        int m_unitarySystemType_Num;
+        bool m_IsUnitarySystem;
+        bool m_IsCoilSystemCoolingDX;
+        bool m_IsCoilSystemCoolingWater;
```

Comment:
3 new bools to designate which system type is being modeled.

Reply:
Hmmmm. If these three `bools` are exclusive (i.e., one and only one of them is true at any time) then the cleaner, less buggy way to implement this would be as an `enum`. Does the fact that you changed this from what was effectively an enum suggest that they are not exclusive? In that case, I may want to use this as a test particle for a "flags" variable pattern, although we can do that outside of this PR.

Reply:
Yes they are exclusive. And will be used as:

    if (this->**m_**IsCoilSystemCoolingDX) {

and this is better/faster?

    enum class SysType = int
    {
        UnitarySystem,
        CoilSystemCoolingDX,
        CoilSystemCoolingWater
    };

    SysType systemType;
 
    if (this->systemType == SysType::CoilSystemCoolingDX) {

Reply:
It is certainly better programming because the `enum` itself forces/implies exclusivity. In general, a fixed set of exclusive options should be represented as an enumeration not as a set of boolean flags. With enumerations you can also handle the different cases using a `switch/case` instead of an `if-else-if` ladder, and that can be faster too especially if you have more than one or two cases that occur relatively frequently. FWIW the syntax is `class enum SysType : int`, and we also define `Unassigned = -1` as the first item of every enumeration and `Num` as the last item, so the declaration here would be:

```
class enum SysType : int 
{
   Unassigned = -1,
   Unitary, 
   CoilCoolingDX,
   CoilCoolingWater,
   Num 
};
```

Notice, I also removed the word `System` from the enumerated values since it is already part of the enumeration type name which is used as a name scope in this case (this is actually the reason we are using `class enum` as opposed to normal `enum`, because the name scoping provides an additional layer of type safety).